datadayessentials/modelling/tests/test_model_manager.py

```diff
@@ -66,3 +67,50 @@ def test_register_model(self):
         assert model_name not in [model for model in self.model_manager.workspace.models]
 
 
+class TestModelCacher(unittest.TestCase):
+    def setUp(self):
+        self.model_name = "model"
+        self.model_version = 1
+        self.cache_directory = Path.home() / "cache"
+        self.model_path = Path.home() / "model"
+        self.model_cache_path = self.cache_directory / f"{self.model_name}-{self.model_version}"
+        self.model_cacher = ModelCacher(self.model_name, self.model_version)
+
+    @patch("datadayessentials.modelling.model_manager.Config")
+    def test_get_model_cache_path(self, mock_config):
+        mock_config.return_value = MagicMock(cache_directory=self.cache_directory)
+        self.assertEqual(self.model_cache_path, self.model_cacher._get_model_cache_path())
+
+    @patch("datadayessentials.modelling.model_manager.Config")
+    def test_is_model_cached(self, mock_config):
+        self._clean_up()
```

Comment:
why running clean up at the start of the function. It should be called at the end of functions to my knowledge